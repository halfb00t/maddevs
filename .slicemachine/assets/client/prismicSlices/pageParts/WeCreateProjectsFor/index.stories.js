import MyComponent from '../../../../../../client/prismicSlices/pageParts/WeCreateProjectsFor';
import SliceZone from 'vue-slicezone'

export default {
  title: 'client/prismicSlices/pageParts/WeCreateProjectsFor'
}


export const _DefaultSlice = () => ({
  components: {
    MyComponent,
    SliceZone
  },
  methods: {
    resolve() {
      return MyComponent
    }
  },
  data() {
    return {
      mock: {"variation":"default-slice","name":"Default slice","slice_type":"we_create_projects_for","items":[],"primary":{"background":"black","animation":"fade-up-right"},"id":"_DefaultSlice"}
    }
  },
  template: '<SliceZone :slices="[mock]" :resolver="resolve" />'
})
_DefaultSlice.storyName = 'Default slice'

export const _WeCreateProjectsForWithMap = () => ({
  components: {
    MyComponent,
    SliceZone
  },
  methods: {
    resolve() {
      return MyComponent
    }
  },
  data() {
    return {
      mock: {"variation":"weCreateProjectsForWithMap","name":"WeCreateProjectsForWithMap","slice_type":"we_create_projects_for","items":[],"primary":{"background":"black","animation":"zoom-in-up"},"id":"_WeCreateProjectsForWithMap"}
    }
  },
  template: '<SliceZone :slices="[mock]" :resolver="resolve" />'
})
_WeCreateProjectsForWithMap.storyName = 'WeCreateProjectsForWithMap'
