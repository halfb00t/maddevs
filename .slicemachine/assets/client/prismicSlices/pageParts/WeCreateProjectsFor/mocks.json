[
  {
    "variation": "default-slice",
    "name": "Default slice",
    "slice_type": "we_create_projects_for",
    "items": [],
    "primary": {
      "background": "black",
      "animation": "fade-up-right"
    }
  },
  {
    "variation": "weCreateProjectsForWithMap",
    "name": "WeCreateProjectsForWithMap",
    "slice_type": "we_create_projects_for",
    "items": [],
    "primary": {
      "background": "black",
      "animation": "zoom-in-up"
    }
  }
]