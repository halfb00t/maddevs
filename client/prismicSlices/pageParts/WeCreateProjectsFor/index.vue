<template>
  <WeCreateProjectsForWithMap
    :slice="slice"
  />
</template>

<script>
// DO NOT USE DEFAULT VARIATION !
import WeCreateProjectsForWithMap
  from '@/prismicSlices/pageParts/WeCreateProjectsFor/variations/WeCreateProjectsForWithMap/WeCreateProjectsForWithMap'

export default {
  name: 'WeCreateProjectsFor',
  components: {
    WeCreateProjectsForWithMap,
  },

  props: {
    slice: {
      type: Object,
      required: true,
      default() {
        return {}
      },
    },
  },
}
</script>
