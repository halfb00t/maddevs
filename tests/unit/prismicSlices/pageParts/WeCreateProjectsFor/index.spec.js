import { shallowMount } from '@vue/test-utils'
import { render, screen } from '@testing-library/vue'
import WeCreateProjectsFor from '@/prismicSlices/pageParts/WeCreateProjectsFor/variations/default-slice/default-slice'

const backgrounds = {
  white: '#ffffff',
  black: '#111213',
  grey: '#f5f7f9',
}

const data = () => ({
  countries: [
    'usa',
    'australia',
  ],
  industriesEntries: [
    ['Content Streaming', 'Transportation'],
    ['E-Commerce', 'AdTech'],
    ['E-Learning', 'FinTech'],
  ],
})

const apiData = {
  background: 'black',
  animation: 'fade-up',
}

const getProps = params => ({
  slice: {
    primary: {
      ...params,
    },
  },
})

describe('WeCreateProjectsFor slice', () => {
  it('should render correctly with data', () => {
    const { container } = render(WeCreateProjectsFor, {
      data,
      props: getProps(apiData),
    })

    expect(screen.getAllByTestId('projects-list__item')).toHaveLength(data().industriesEntries.length)
    expect(screen.getAllByTestId('flags-list__item')).toHaveLength(data().countries.length)
    expect(container).toMatchSnapshot()
  })

  it('should render correctly without data', () => {
    const { container } = render(WeCreateProjectsFor, {
      data: () => ({
        countries: [],
        industriesEntries: [],
      }),
      props: getProps({}),
    })

    expect(screen.queryByTestId('projects-list__item')).toBeNull()
    expect(screen.queryByTestId('flags-list__item')).toBeNull()
    expect(container).toMatchSnapshot()
  })

  describe('data-aos animation attribute', () => {
    it(`should be '${apiData.animation}'`, () => {
      render(WeCreateProjectsFor, {
        data,
        props: getProps(apiData),
      })

      expect(screen.getByTestId('create-projects-slice__content').getAttribute('data-aos')).toBe(apiData.animation)
    })

    it('should be null', () => {
      render(WeCreateProjectsFor, {
        data,
        props: getProps({}),
      })

      expect(screen.getByTestId('create-projects-slice__content').getAttribute('data-aos')).toBeNull()
    })
  })

  describe('sliceBackground computed method', () => {
    it(`should return ${backgrounds.white} (white)`, () => {
      const wrapper = shallowMount(WeCreateProjectsFor, {
        propsData: getProps({
          ...apiData,
          background: 'white',
        }),
      })

      expect(wrapper.vm.sliceBackground).toBe(backgrounds.white)
    })

    it(`should return ${backgrounds.grey} (grey)`, () => {
      const wrapper = shallowMount(WeCreateProjectsFor, {
        propsData: getProps({
          ...apiData,
          background: 'grey',
        }),
      })

      expect(wrapper.vm.sliceBackground).toBe(backgrounds.grey)
    })

    it(`should return ${backgrounds.black} (black)`, () => {
      const wrapper = shallowMount(WeCreateProjectsFor, {
        propsData: getProps({
          ...apiData,
          background: 'black',
        }),
      })

      expect(wrapper.vm.sliceBackground).toBe(backgrounds.black)
    })

    it('should return null', () => {
      const wrapper = shallowMount(WeCreateProjectsFor, {
        propsData: getProps({
          ...apiData,
          background: 'unknown',
        }),
      })

      expect(wrapper.vm.sliceBackground).toBeNull()
    })
  })
})
