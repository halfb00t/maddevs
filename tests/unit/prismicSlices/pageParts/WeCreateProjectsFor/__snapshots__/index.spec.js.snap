// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeCreateProjectsFor slice should render correctly with data 1`] = `
<div>
  <section
    class="create-projects-slice"
    style="background-color: rgb(17, 18, 19);"
  >
    <div
      class="container"
    >
      <div
        class="create-projects-slice__content aos-init aos-animate"
        data-aos="fade-up"
        data-testid="create-projects-slice__content"
      >
        <h2
          class="create-projects-slice__title"
        >
          
        We create 
          <br />
           IT projects for
      
        </h2>
         
        <div
          class="create-projects-slice__projects projects-list"
        >
          <div
            class="projects-list__item"
            data-testid="projects-list__item"
          >
            <p
              class="projects-list__item-paragraph"
            >
              
            Content Streaming
            
              <br />
              
            Transportation
          
            </p>
          </div>
          <div
            class="projects-list__item"
            data-testid="projects-list__item"
          >
            <p
              class="projects-list__item-paragraph"
            >
              
            E-Commerce
            
              <br />
              
            AdTech
          
            </p>
          </div>
          <div
            class="projects-list__item"
            data-testid="projects-list__item"
          >
            <p
              class="projects-list__item-paragraph"
            >
              
            E-Learning
            
              <br />
              
            FinTech
          
            </p>
          </div>
        </div>
         
        <ul
          class="create-projects-slice__flags flags-list"
        >
          <li
            class="flags-list__item"
            data-testid="flags-list__item"
          >
            <img
              alt="usa"
              height="32"
              src="[object Object]"
              width="32"
            />
          </li>
          <li
            class="flags-list__item"
            data-testid="flags-list__item"
          >
            <img
              alt="australia"
              height="32"
              src="[object Object]"
              width="32"
            />
          </li>
        </ul>
      </div>
    </div>
  </section>
</div>
`;

exports[`WeCreateProjectsFor slice should render correctly without data 1`] = `
<div>
  <section
    class="create-projects-slice"
  >
    <div
      class="container"
    >
      <div
        class="create-projects-slice__content"
        data-testid="create-projects-slice__content"
      >
        <h2
          class="create-projects-slice__title"
        >
          
        We create 
          <br />
           IT projects for
      
        </h2>
         
        <div
          class="create-projects-slice__projects projects-list"
        />
         
        <ul
          class="create-projects-slice__flags flags-list"
        />
      </div>
    </div>
  </section>
</div>
`;
